drivers/overlay/overlay.go

```diff
@@ -562,8 +562,35 @@ func (d *Driver) Get(id, mountLabel string) (_ string, retErr error) {
 		return "", err
 	}
 
-	newlowers := ""
+	// absLowers is the list of lowers as absolute paths, which works well with additional stores.
+	absLowers := ""
+	// relLowers is the list of lowers as paths relative to the driver's home directory.
+	relLowers := ""
+
+	// Check if $link/../diff{1-*} exist.  If they do, add them, in order, as the front of the lowers
+	// lists that we're building.
+	link, err := ioutil.ReadFile(path.Join(dir, "link"))
+	if err != nil {
+		return "", err
+	}
+	diffN := 1
+	_, err = os.Stat(filepath.Join(dir, nameWithSuffix("diff", diffN)))
+	for err == nil {
+		if absLowers == "" {
+			absLowers = filepath.Join(dir, nameWithSuffix("diff", diffN))
+			relLowers = dumbJoin(string(link), "..", nameWithSuffix("diff", diffN))
+		} else {
+			absLowers = absLowers + ":" + filepath.Join(dir, nameWithSuffix("diff", diffN))
+			relLowers = relLowers + ":" + dumbJoin(string(link), "..", nameWithSuffix("diff", diffN))
```

Comment:
Nit: this would probably be a bit more convenient, and easier to follow, with all the `*lowers` variables a `[]string`, only joining them with colons at the very end.

Reply:
Yeah, I was aiming for keeping the list of changes wrt moby's version smaller in case we rebase again, but we have tests for that, and that cleanup makes sense.